src/main/java/org/broadinstitute/consent/http/service/MetricsService.java

```diff
@@ -160,4 +194,54 @@ public String getHeaderRow(Type type) {
     }
     return election;
   }
+
+  public DatasetMetrics generateDatasetMetrics(Integer datasetId) {
+
+    DatasetMetrics metrics = new DatasetMetrics();
+
+    //get datasetDTO with properties and data use restrictions
+    Set<DatasetDTO> datasets = dataSetDAO.findDatasetDTOWithPropertiesByDatasetId(datasetId);
+    if (datasets == null || datasets.isEmpty()) {
+       throw new NotFoundException("Dataset with specified ID does not exist.");
+    }
+
+    //find dars with the given datasetId in their list of datasetIds, datasetId is a String so it can be converted to jsonb in query
+    //convert all dars into smaller objects that only contain the information needed
+    List<DataAccessRequest> dars = dataAccessRequestDAO.findAllDataAccessRequestsForDatasetMetrics(Integer.toString(datasetId));
+    List<DarMetricsSummary> darInfo = dars.stream().map(dar -> 
+      new DarMetricsSummary(dar.getUpdateDate(), dar.data.getProjectTitle(), dar.data.getDarCode(), dar.data.getNonTechRus(), findPI(dar.userId), dar.getReferenceId()))
```

Comment:
Use getters for fields. This would be a lot cleaner if the constructor took a Dar instead of individual fields. In the constructor, you could add null checks for safety, etc.